packages/cli-repl/src/arg-parser.ts

```diff
@@ -137,28 +146,37 @@ export function parseCliArgs(args: string[]): (CliOptions & { smokeTests: boolea
   const programArgs = args.slice(2);
   i18n.setLocale(getLocale(programArgs, process.env));
 
-  const parsed = parser(programArgs, OPTIONS);
-  parsed._ = parsed._.filter(arg => {
-    if (arg === START) {
-      return false;
-    }
-    // mongosh is currently not taking any numerical positional arguments.
-    if (typeof arg === 'string' && !arg.startsWith('-')) {
-      return true;
+  const parsed = parser(programArgs, OPTIONS) as unknown as CliOptions & {
+    smokeTests: boolean;
+    _?: string[];
+    file?: string[];
+  };
+  const positionalArguments = parsed._ ?? [];
+  for (const arg of positionalArguments) {
+    if (arg.startsWith('-')) {
+      throw new Error(
+        `  ${clr(i18n.__(UNKNOWN), ['red', 'bold'])} ${clr(String(arg), 'bold')}
+        ${USAGE}`
+      );
```

Comment:
Not really related to this change, but I was figuring out what case this is handling and found an interesting uhhh... bug? related to how yargs parser works I guess

This exists with `unknown option` error

```
mongosh --------foobar
```

This doesn't exit, the option is parsed by yargs to `--------help: true` so no help is printed

```
mongosh ----------help
```

Reply:
That”s … interesting, yeah. I feel like this might even qualify as yargs bug

Reply:
Yeah, true, I'll take a look if they have something like this already submitted and will open an issue/send a patch or something. I think it's a big enough corner case for us to not really worry about 🤔